packages/dd-trace/src/debugger/devtools_client/snapshot/processor.js

```diff
@@ -195,6 +201,9 @@ function toMap (type, pairs, maxLength) {
       // This can be skipped and we can go directly to its children, of which
       // there will always be exactly two, the first containing the key, and the
       // second containing the value of this entry of the Map.
+      if (Object.hasOwn(value, timeBudgetSym)) {
```

Comment:
I think we can reduce the snapshot size by simply filter those items out and put noCaptureReason: 'timeout' on the map/collection object itself. same as size capture limit.

Reply:
I agree. I didn't like this bit either. I took a second look and found a better implementation that addresses your concern.